src/Docker/Runner.php

```diff
@@ -202,12 +202,20 @@ private function runRow(
         $workingDirectory = new WorkingDirectory($temp->getTmpFolder(), $this->loggersService->getLog());
         $usageFile->setDataDir($workingDirectory->getDataDir());
 
+        $tokenInfo = $this->clientWrapper->getBasicClient()->verifyToken();
+        $jobScopedEncryptor = new JobScopedEncryptor(
+            $this->encryptor,
+            $jobDefinition->getComponentId(),
+            $tokenInfo['owner']['id'],
+            $jobDefinition->getConfigId()
+        );
+
         $configData = $jobDefinition->getConfiguration();
-        $authorization = new Authorization($this->oauthClient3, $this->encryptorFactory->getEncryptor(), $component->getId());
-        $imageParameters = $this->encryptorFactory->getEncryptor()->decrypt($component->getImageParameters());
+        $authorization = new Authorization($this->oauthClient3, $jobScopedEncryptor, $component->getId());
+
         $configFile = new ConfigFile(
             $workingDirectory->getDataDir(),
-            $imageParameters,
+            $this->encryptor->decryptForComponent($component->getImageParameters(), $component->getId()),
```

Comment:
Tohle se predtim dekryptovalo jakoby tim jobScopedEncryptorem, ale logicky jsme si rikal, ze image parametry u komponenty muzou byt sifrovany jen na komponentu, ne? Nemuzou byt projekt/config specific.

Reply:
nemelo by to nic zkazit.. radeji ho tam teda hodim